libraries/api-documenter/src/Domifier.ts

```diff
@@ -0,0 +1,225 @@
+    return {
+      kind: 'note-box',
+      elements: textElements
+    } as IDomNoteBox;
+  }
+
+  public static createNoteBoxFromText(text: string): IDomNoteBox {
+    return Domifier.createNoteBox(Domifier.createTextElements(text));
+  }
+
+  public static createTableRow(cellValues: DomBasicText[][] | undefined = undefined): IDomTableRow {
+    const row: IDomTableRow = {
+      kind: 'table-row',
+      cells: []
+    };
+
+    if (cellValues) {
+      for (const cellValue of cellValues) {
+        const cell: IDomTableCell = {
+          kind: 'table-cell',
+          elements: cellValue
+        };
+        row.cells.push(cell);
+      }
+    }
+
+    return row;
+  }
+
+  public static createTable(headerCellValues: DomBasicText[][] | undefined = undefined): IDomTable {
+    let header: IDomTableRow | undefined = undefined;
+    if (headerCellValues) {
+      header = Domifier.createTableRow(headerCellValues);
+    }
+    return {
+      kind: 'table',
+      header: header,
+      rows: []
+    } as IDomTable;
+  }
+
+  public static createPage(title: string, docId: string): IDomPage {
+    return {
+      kind: 'page',
+      title: title,
+      docId: docId,
+      elements: []
+    } as IDomPage;
+  }
+
+  public static renderDocElements(docElements: IDocElement[] | undefined): DomBasicText[] {
+    if (!docElements) {
+      return [];
+    }
+
+    const result: DomBasicText[] = [];
+
+    for (const docElement of docElements || []) {
+      switch (docElement.kind) {
+        case 'textDocElement':
+          const textDocElement: ITextElement = docElement as ITextElement;
+          result.push(...Domifier.createTextElements(textDocElement.value));
+          break;
+        case 'linkDocElement':
+          const linkDocElement: ILinkDocElement = docElement as ILinkDocElement;
+          if (linkDocElement.referenceType === 'code') {
+            let linkText: string|undefined = linkDocElement.value;
+            if (!linkText) {
+              linkText = linkDocElement.exportName;
+              if (linkDocElement.memberName) {
+                linkText += '.' + linkDocElement.memberName;
+              }
+            }
+            result.push(
+              Domifier.createDocLinkFromText(linkText,
+                RenderingHelpers.getDocId(linkDocElement.packageName || '', linkDocElement.exportName,
+                  linkDocElement.memberName)
+              )
+            );
+          } else {
+            result.push(
+              {
+                kind: 'web-link',
+                elements: Domifier.createTextElements(linkDocElement.value || linkDocElement.targetUrl),
+                targetUrl: linkDocElement.targetUrl
+              } as IDomWebLink
+            );
+          }
+          break;
+        case 'seeDocElement':
+          const seeDocElement: ISeeDocElement = docElement as ISeeDocElement;
+          result.push(
+            ...Domifier.createTextElements('see ') // @todo
```

Comment:
>// @todo [](start = 51, length = 8)

do we have an issue # ?

Reply:
I don't want to track that.  I'll just add a comment to the code saying the implementation could be improved later.

---
In reply to: [135167431](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135167431) [](ancestors = 135167431)